Tests/AsyncHTTPClientTests/TransactionTests.swift

```diff
@@ -541,16 +541,26 @@ final class TransactionTests: XCTestCase {
 // tasks. Since we want to wait for things to happen in tests, we need to `async let`, which creates
 // implicit tasks. Therefore we need to wrap our iterator struct.
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-actor SharedIterator<Iterator: AsyncIteratorProtocol> {
-    private var iterator: Iterator
+actor SharedIterator<Wrapped: AsyncSequence> where Wrapped.Element: Sendable {
+    private var wrappedIterator: Wrapped.AsyncIterator
+    private var nextCallInProgress: Bool = false
 
-    init(_ iterator: Iterator) {
-        self.iterator = iterator
+    init(_ sequence: Wrapped) {
```

Comment:
Iterator is now created in the actor instead of outside the actor. This is necessary because `AsyncIterator`s are often non-`Sendable` but `AsyncSequence`s themself are often `Sendable`. Passing in an non-`Sendable ` value produces a warning with Swift 5.8 (Swift Development Snapshot 2022-08-18 (a)) e.g.:
```
actor Actor<Element> {
    init(_ element: Element) {}
}
class NonSendableClass {}

let nonSendable = NonSendableClass()
_ = Actor(nonSendable) // warning: Non-sendable type 'NonSendableClass' passed in call to nonisolated initializer 'init(_:)' cannot cross actor boundary
````
I'm not 100% sure if this warning is correct and need to think a bit more about this warning. However, the change doesn't hurt either.